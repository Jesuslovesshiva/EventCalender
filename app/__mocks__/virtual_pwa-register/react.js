import {useState} from 'react'

const useRegisterSW = jest.fn(() => {
  const [needRefresh, setNeedRefresh] = useState(true)

  return {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker: () => {}
  }
})

// eslint-disable-next-line import/prefer-default-export
export {useRegisterSW}
