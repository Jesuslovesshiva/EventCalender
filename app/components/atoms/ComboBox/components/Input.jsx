import classNames from 'classnames'
import PropTypes from 'prop-types'
import {components} from 'react-select'

/* eslint-disable react/jsx-props-no-spreading */
const Input = ({children, className, ...props}) => (
  <components.Input
    className={classNames('!m-0 !p-0 !text-gray-dark', className)}
    {...props}
  >
    {children}
  </components.Input>
)
/* eslint-enable */

Input.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
}

Input.defaultProps = {
  children: null,
  className: null
}

export default Input
