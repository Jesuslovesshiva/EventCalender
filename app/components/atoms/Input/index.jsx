import classNames from 'classnames'
import {omit} from 'lodash'
import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {useCallback, useState} from 'react'

import WarningAmberIcon from '@/components/atoms/Icon/WarningAmber'
import useSyncState from '@/hooks/use-sync-state'
import useTranslation from '@/hooks/use-translation'
import propTypes from '@/utils/prop-types'

const Input = ({component: Component, children, ...props}) => {
  const placeholder = useTranslation(props.placeholder)
  const [value, setValue] = useSyncState(props.value)
  const [hasFocus, setHasFocus] = useState(props.autoFocus)
  const onFocus = useCallback(() => setHasFocus(true), [setHasFocus])
  const onBlur = useCallback(() => setHasFocus(false), [setHasFocus])
  const onChange = useCallback(
    event => {
      props.onChange(event.target.value)
      setValue(event.target.value)
    },
    [props.onChange, setValue]
  )

  const passThroughProps = omit(props, [
    'className',
    'hasError',
    'inputRef',
    'isDisabled',
    'isRequired',
    'onChange',
    'placeholder',
    'startIcon',
    'value'
  ])

  return (
    <Component
      className={classNames(
        props.className,
        'border rounded transition inline-flex items-center relative overflow-hidden',
        /* eslint-disable sort-keys */
        {
          'border-gray-medium hover:border-white hover:bg-silver hover:shadow':
            !hasFocus,
          'border-white bg-white shadow': hasFocus,
          'border-primary hover:border-primary': props.hasError,
          'border-gray-medium hover:bg-transparent hover:shadow-none cursor-not-allowed':
            props.isDisabled,
          'hover:border-gray-medium': !props.isDisabled && props.hasError
        }
        /* eslint-enable sort-keys */
      )}
    >
      {Boolean(props.startIcon || props.hasError) && (
        <div
          className={classNames(
            'flex absolute left-0 top-[7px] ml-2.5 mr-[-7px]',
            'text-lg pointer-events-none',
            {
              'text-primary': props.hasError
            }
          )}
        >
          {Boolean(props.startIcon && !props.hasError) && props.startIcon}
          {Boolean(props.hasError) && <WarningAmberIcon />}
        </div>
      )}

      <input
        className={classNames(
          'border-0 px-[0.6875rem] py-1.5 w-full',
          'bg-transparent text-gray-dark placeholder-gray-medium focus:outline-none',
          /* eslint-disable sort-keys */
          {
            'text-base': props.type !== 'number',
            'number-input font-button-text peer pl-[0.5625rem] pr-0.5':
              props.type === 'number',
            'pl-8':
              props.hasError || (props.startIcon && props.type !== 'number'),
            'pl-7':
              (props.hasError || props.startIcon) && props.type === 'number',
            'text-gray-medium cursor-not-allowed': props.isDisabled
          }
          /* eslint-enable sort-keys */
        )}
        disabled={props.isDisabled}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        required={props.isRequired}
        value={value}
        placeholder={placeholder}
        ref={props.inputRef}
        {...passThroughProps}
      />

      {children}
    </Component>
  )
}

Input.propTypes = {
  autoFocus: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
  component: PropTypes.elementType,
  hasError: PropTypes.bool,
  inputRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.any})
  ]),
  isDisabled: PropTypes.bool,
  isRequired: PropTypes.bool,
  max: PropTypes.string,
  min: PropTypes.string,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  placeholder: propTypes.Translatable,
  startIcon: PropTypes.node,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

Input.defaultProps = {
  autoFocus: false,
  children: null,
  className: null,
  component: 'label',
  hasError: false,
  inputRef: null,
  isDisabled: false,
  isRequired: false,
  max: null,
  min: null,
  onChange: noop,
  onKeyDown: noop,
  placeholder: null,
  startIcon: null,
  type: null,
  value: ''
}

export default Input
