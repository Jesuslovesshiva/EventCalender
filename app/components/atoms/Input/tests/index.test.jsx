import {createEvent, fireEvent} from '@testing-library/react'

import render from '@/utils/test-utils/custom-render'

import Input from '../index'

describe('Input', () => {
  it('should match the snapshot', () => {
    const {container} = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should change styling on focus', () => {
    const {container} = render(<Input />)
    const input = container.querySelector('input')

    fireEvent.focus(input)
    expect(container.firstChild).toHaveClass('border-white bg-white')
    fireEvent.blur(input)
    expect(container.firstChild).not.toHaveClass('border-white bg-white')
  })

  it('should update input value on value prop change', () => {
    const {rerender, container} = render(<Input />)
    const input = container.querySelector('input')

    fireEvent.change(input, {target: {value: 'foo'}})
    expect(input.value).toBe('foo')
    rerender(<Input value="bar" />)
    expect(input.value).toBe('bar')
  })

  it('should call onChange prop on change', () => {
    const onChange = jest.fn()
    const {container} = render(<Input onChange={onChange} />)
    const input = container.querySelector('input')

    expect(onChange).not.toHaveBeenCalled()
    fireEvent.change(input, {target: {value: 'foo'}})
    expect(onChange).toHaveBeenCalledWith('foo')
  })

  it('should not prevent non-numbers to be entered for text input', () => {
    const {container} = render(<Input type="text" />)
    const input = container.querySelector('input')

    const keyDownEvent = createEvent.keyDown(input, {key: 'a'})
    fireEvent(input, keyDownEvent)
    expect(keyDownEvent.defaultPrevented).toBe(false)
  })
})
