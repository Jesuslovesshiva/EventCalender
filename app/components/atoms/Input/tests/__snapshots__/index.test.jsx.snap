// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
<label
  class="border rounded transition inline-flex items-center relative overflow-hidden border-gray-medium hover:border-white hover:bg-silver hover:shadow"
>
  <input
    class="border-0 px-[0.6875rem] py-1.5 w-full bg-transparent text-gray-dark placeholder-gray-medium focus:outline-none text-base"
    value=""
  />
</label>
`;
